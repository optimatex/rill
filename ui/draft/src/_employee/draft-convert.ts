export { convertToHTML, convertFromHTML } from 'draft-convert';
