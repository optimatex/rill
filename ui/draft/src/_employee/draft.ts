export { Editor, EditorState, convertToRaw } from 'draft-js';
