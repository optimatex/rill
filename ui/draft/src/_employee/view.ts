export {
  React,
  useState,
  useCallback,
  css,
  styled,
  createGlobalStyle,
} from '@rill/view';
