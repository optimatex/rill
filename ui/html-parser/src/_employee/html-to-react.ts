export { Parser, ProcessNodeDefinitions } from 'html-to-react';
