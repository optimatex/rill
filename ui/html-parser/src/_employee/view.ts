export { React, RC } from '@rill/view';
