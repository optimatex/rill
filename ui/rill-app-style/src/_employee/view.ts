export { createGlobalStyle, css } from '@rill/view';
