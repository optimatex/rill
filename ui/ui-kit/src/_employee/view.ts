export {
  React,
  useState,
  useCallback,
  // styled-components
  styled,
  // Lens
  // getTestId,
  // Types
  FC,
  RC,
  ReactNode,
  LegacyRef,
  // PropTypes
  IContainerProps,
  // Event types
  FormEvent,
  MouseEvent,
} from '@rill/view';
