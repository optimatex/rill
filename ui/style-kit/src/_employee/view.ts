import { css as originalCss } from '@rill/view';

export const css = originalCss;
