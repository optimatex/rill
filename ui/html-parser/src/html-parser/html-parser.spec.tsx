import { React } from '_employee/view';
import { render, mockString, mockCombinedRenderProps } from 'test';

import { htmlParser } from '.';

describe('Text Editor Container', () => {
  it('should render without errors', () => {
    const { getByRole } = render(
      <article>{htmlParser(mockString, mockCombinedRenderProps)}</article>,
    );
    const el = getByRole('article');

    expect(el).toMatchSnapshot();
  });

  it('should pass proper props to components', () => {
    const mockRenderParagraph = jest.fn();
    const mockRenderImage = jest.fn();
    const mockRenderYoutubes = jest.fn();
    const renders = {
      renderParagraph: mockRenderParagraph,
      renderImages: mockRenderImage,
      renderYoutubes: mockRenderYoutubes,
    };
    render(<article>{htmlParser(mockString, renders)}</article>);

    expect(mockRenderParagraph).toHaveBeenCalled();
    expect(mockRenderParagraph.mock.calls[0]).toMatchSnapshot();

    expect(mockRenderImage).toHaveBeenCalled();
    expect(mockRenderImage.mock.calls[0]).toEqual([
      [
        {
          Height: 0,
          Id: 38318,
          ImageId: '96f90d02-6a3d-4535-961d-a2d4e2571e6b',
          Width: 0,
        },
      ],
      'grid',
    ]);

    expect(mockRenderYoutubes).toHaveBeenCalled();
    expect(mockRenderYoutubes.mock.calls[0]).toEqual([
      [
        {
          Id: 2073,
          Source: 'https://www.youtube.com/watch?v=VhkfnPVQyaY',
          YouTubeId: 'VhkfnPVQyaY',
        },
      ],
      'grid',
    ]);
  });
});
