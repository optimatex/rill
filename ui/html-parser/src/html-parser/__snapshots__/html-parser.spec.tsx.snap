// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Editor Container should pass proper props to components 1`] = `
Array [
  Array [
    "Египетский фараон Джосера известный тем, что активно искал  артефакты и знания богов. Одним из результатов этих поисков стала огромная коллекция предметов, найдена в галереях под его пирамидой. ",
  ],
  1,
]
`;

exports[`Text Editor Container should render without errors 1`] = `
<article>
  

  <p>
    Египетский фараон Джосера известный тем, что активно искал  артефакты и знания богов. Одним из результатов этих поисков стала огромная коллекция предметов, найдена в галереях под его пирамидой. 
  </p>
  <figure
    class="contentMedia "
  >
    <img
      src="96f90d02-6a3d-4535-961d-a2d4e2571e6b"
    />
  </figure>
  <p>
    Коллекцию составляли преимущественно чаши и вазы. Но среди них были сосуды маленького размера из полупрозрачного камня, по форме напоминающие лампочки или химические колбы. Так же там присутствовали предметы совершенно непонятного назначения, напоминающие лопасти или другие технологические детали.
  </p>
  <figure
    class="contentMedia "
  >
    <img
      src="6411477f-e6bc-427d-84f5-1d557b2cc82a"
    />
  </figure>
  <p>
    Большинство предметы изготовлено в невероятно высоком качестве, из точным соблюдением геометрических пропорций и фигур. Некоторые стенки сосудов толщиной всего в 1-2 мм., другие имеют надписи толщиной в 0.15 мм. Все это указывает на следы машинной обработки
  </p>
  

  <figure
    class="contentMedia "
  >
    <iframe />
  </figure>
  <figure
    class="contentMedia "
  >
    <iframe />
  </figure>
  <figure
    class="contentMedia "
  >
    <iframe />
  </figure>
  <figure
    class="contentMedia "
  >
    <iframe />
  </figure>
  <figure
    class="contentMedia "
  >
    <iframe />
  </figure>
  <figure
    class="contentMedia "
  >
    <iframe />
  </figure>
  <figure
    class="contentMedia "
  >
    <iframe />
  </figure>
  

</article>
`;
